pygsti/modelmembers/povms/tppovm.py

```diff
@@ -56,3 +59,35 @@ def __reduce__(self):
 
         return (TPPOVM, (effects, self.evotype, self.state_space, True),
                 {'_gpindices': self._gpindices, '_submember_rpindices': self._submember_rpindices})
+    
+    @property
+    def dim(self):
+        effect = next(iter(self.values()))
+        return effect.dim
+    
+    def to_vector(self):
+        effect_vecs = []
+        for i, (lbl, effect) in enumerate(self.items()):
+            if lbl != self.complement_label:
+                assert isinstance(effect, _FullPOVMEffect)
+                effect_vecs.append(effect.to_vector())
+            else:
+                assert i == len(self) - 1
+        vec = _np.concatenate(effect_vecs)
+        return vec
+
+    def stateless_data(self) -> Tuple[int, int]:
+        dim1 = len(self)
+        dim2 = self.dim
+        return (dim1, dim2)
+
+    @staticmethod
+    def torch_base(sd: Tuple[int, int], t_param: _Torchable.Tensor) -> _Torchable.Tensor:
+        torch = _Torchable.torch_handle
+        num_effects, dim = sd
+        first_basis_vec = torch.zeros(size=(1, dim), dtype=torch.double)
+        first_basis_vec[0,0] = dim ** 0.25
+        t_param_mat = t_param.reshape((num_effects - 1, dim))
+        t_func = first_basis_vec - t_param_mat.sum(axis=0, keepdim=True)
+        t = torch.row_stack((t_param_mat, t_func))
+        return t
```

Comment:
@coreyostrove, @sserita:  is the ``dim ** 0.25`` scale appropriate when the underlying vector space is something other than "tensor product of qubit space"?

Reply:
The verdict is that TPPOVM already restricted to the Pauli-product or Gell-Mann bases, in which situation this scaling is appropriate.